### PR TITLE
feat: broker

### LINKED ISSUE #63: contract to supply and deposit LP in one step
As an end user, I need to be able to get an LP and deposit it into a "meta" farm bot in a single step.

Eng:
the flow should include the following steps:
- user approves Farm Investor (not attached to this naming) for X of token A
- user approves Farm Investor for Y of token B (equal value)
- user calls "invest" method on farm investor
- farm investor takes token A and token B from user
- farm investor approves liquidity router for token A and token B
- farm investor adds liquidity and gets LP in return
- farm investor deposits LP into farm bot and gets RFP in return
- farm investor sends RFP AND ANY UNUSED token A/B to the end user

AC:
- the above contract is implemented and unit tested
- some e2e test is run on alfajores or mainnet

### PR DESCRIPTION
**Pull request by cajubelt**
allows you to get LP deposit them into a farm bot for RFP in a single transaction

fixes #63

### CODE REVIEW

File: contracts/mocks/MockFarmBot.sol [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import "../ubeswap/contracts/uniswapv2/interfaces/IUniswapV2Router02.sol";
+import "../ubeswap/contracts/uniswapv2/interfaces/IUniswapV2Router02SwapOnly.sol";
+import "./MockLPToken.sol";
+
+/**
+* MockFarmBot has the bare minimum functionality of a farm bot for testing RevoFPBroker.
+*/
+contract MockFarmBot is MockERC20 {
+    IUniswapV2Router02SwapOnly public swapRouter; // address to use for router that handles swaps
```
Comment (jophish) [inline contracts/mocks/MockFarmBot.sol:11]: nit: `swapRouter` isn't actually necessary for the broker contract, right?

File: contracts/library/UniswapRouter.sol [hunk 2]
```diff
@@ -78,13 +78,13 @@ library UniswapRouter {
         uint256 amount0Min,
         uint256 amount1Min,
         uint256 deadline
-    ) internal {
+    ) internal returns (uint amountA, uint amountB, uint liquidity) {
```
Comment (jophish) [inline contracts/library/UniswapRouter.sol:81]: nice change!

**Review by jophish — APPROVED** [review]
Didn't look super closely at the contents of the `/scripts` directory, but this looks awesome.

File: contracts/mocks/MockFarmBot.sol (hunk 1 above)
Comment (cajubelt) [reply contracts/mocks/MockFarmBot.sol:11]: an address is given in the constructor so I went ahead and saved it

**Pull request merged**